unstructured/ingest/connector/fsspec.py

```diff
@@ -111,13 +127,76 @@ def get_file(self):
             **self.config.get_access_kwargs(),
         )
         logger.debug(f"Fetching {self} - PID: {os.getpid()}")
-        fs.get(rpath=self.remote_file_path, lpath=self._tmp_download_file().as_posix())
+        try:
+            fs.get(rpath=self.remote_file_path, lpath=self._tmp_download_file().as_posix())
+        except FileNotFoundError as e:
+            logger.error("Failed to find file in remote source.")
+            logger.error(e)
+
+    def _catch_not_implemented(self, method, **kwargs) -> Optional[str]:
+        try:
+            value: datetime = method(self.remote_file_path, kwargs)
+        except NotImplementedError:
+            return None
+        return value.isoformat()
+
+    @cached_property
+    @requires_dependencies(["fsspec"])
+    def file_metadata(self):
+        """Fetches file metadata from the current filesystem."""
+        from fsspec import AbstractFileSystem, get_filesystem_class
+
+        fs: AbstractFileSystem = get_filesystem_class(self.config.protocol)(
+            **self.config.get_access_kwargs(),
+        )
+        date_created = self._catch_not_implemented(fs.created)
+        date_modified = self._catch_not_implemented(fs.modified)
+        version = str(fs.checksum(self.remote_file_path))
+        source_url = self._catch_not_implemented(
+            fs.sign(self.remote_file_path),
+            expiration=SIGNED_URL_EXPIRATION,
+        )
```

Comment:
I'm really confused here. _catch_not_implemented is assuming we're getting a date, right?